### PR TITLE
Gdr 2769

### PR DESCRIPTION
**Pull request by gladkia**
# Description
Refactor of  `set_unique_*` functions
## What changed?
Make functions work for `DFrame` and `data.table` inputs. (The functions, for the latter, were not working properly.)
Related JIRA issue: https://jira.gene.com/jira/browse/GDR-2769

## Why was it changed?

# Checklist for sustainable code base
- [X] I added tests for any code changed/added
- [X] I added documentation for any code changed/added
- [X] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [X] Package version bumped
- [X] Changelog updated

# Screenshots (optional)

### CODE REVIEW

File: tests/testthat/test-standardize_MAE.R [hunk 1]
```diff
@@ -138,7 +138,7 @@ test_that("set_unique_cl_names_dt and set_unique_drug_names_dt works correctly",
   res_8 <- set_unique_cl_names_dt(dt)
   expect_equal(res_7, dt)
   expect_false(identical(res_8, dt))
-  expect_equal(length(unique(res_8$CellLineName)), 5)
```
Comment (gladkia) [inline tests/testthat/test-standardize_MAE.R:141]: IMO there was a bug in the previous logic, which seems to be fixed in the new version.

File: R/standardize_MAE.R [hunk 2]
```diff
@@ -278,6 +278,44 @@ refine_rowdata <- function(rd, se, default_v = "Undefined") {
...
+#' @return fixed input table with unique primary field in the table
+#' @examples
+#' col_data <- S4Vectors::DataFrame(CellLineName = c("ID1", "ID1"), clid = c("C1", "C2"))
+#' col_data <- set_unique_names_dt(col_data, primary_name = "CellLineName", secondary_name = "clid")
+#' @keywords standardize_MAE
+#' 
+set_unique_names_dt <- function(dt, primary_name, secondary_name, sep = " ") {
+  
+  stopifnot(any(
+    inherits(col_data, "data.table") ||
```
Comment (bczech) [inline R/standardize_MAE.R:299]:
```suggestion
    inherits(dt, "data.table") ||
```
There is no col_data variable define within this function

File: R/standardize_MAE.R [hunk 3]
```diff
@@ -278,6 +278,44 @@ refine_rowdata <- function(rd, se, default_v = "Undefined") {
...
+      inherits(col_data, "DFrame") || inherits(col_data, "data.frame")
```
Comment (bczech) [inline R/standardize_MAE.R:300]:
```suggestion
      inherits(dt, "DFrame") || inherits(dt, "data.frame")
```

Comment (bczech) [reply R/standardize_MAE.R:300]: BTW, maybe we could use checkmate instead?

```
  checkmate::assert(
    checkmate::check_class(dt, "data.table"),
    checkmate::check_class(dt, "DFrame"),
    checkmate::check_class(dt, "data.frame")
  )
```

File: R/standardize_MAE.R [hunk 4]
```diff
@@ -278,6 +278,44 @@ refine_rowdata <- function(rd, se, default_v = "Undefined") {
...
+#' @param sep string with separator added before suffix
+#' @return fixed input table with unique primary field in the table
+#' @examples
+#' col_data <- S4Vectors::DataFrame(CellLineName = c("ID1", "ID1"), clid = c("C1", "C2"))
+#' col_data <- set_unique_names_dt(col_data, primary_name = "CellLineName", secondary_name = "clid")
+#' @keywords standardize_MAE
+#' 
+set_unique_names_dt <- function(dt, primary_name, secondary_name, sep = " ") {
+  
+  stopifnot(any(
```
Comment (bczech) [inline R/standardize_MAE.R:298]: `any` is not necessary when `||` is used

File: R/standardize_MAE.R [hunk 5]
```diff
@@ -406,43 +423,31 @@ set_unique_drug_names <- function(se) {
...
+  checkmate::assert_character(secondary_names) 
   
-  drug_columns <- intersect(unlist(get_env_identifiers(c("drug_name", "drug_name2", "drug_name3"), simplify = FALSE)),
-                            names(row_data))
-  gnumber_columns <- intersect(unlist(get_env_identifiers(c("drug", "drug2", "drug3"), simplify = FALSE)),
-                               names(row_data))
+  primary_names <- intersect(primary_names, names(dt))
+  secondary_names <- intersect(secondary_names, names(dt))
+ 
+  stopifnot(NROW(primary_names) == NROW(secondary_names))
```
Comment (bczech) [inline R/standardize_MAE.R:437]:
```suggestion
  checkmate::assert_true(NROW(primary_names) == NROW(secondary_names))
```

File: R/standardize_MAE.R (hunk 4 above)
Comment (bczech) [reply R/standardize_MAE.R:298]: btw, can we use checkmate here as well, e.g.

checkmate::assert(
  checkmate::check_class(dt, "data.table"),
  checkmate::check_class(dt, "DFrame"),
  checkmate::check_class(dt, "data.frame")
)

**Review by bczech — APPROVED** [review]

File: R/standardize_MAE.R [hunk 6]
```diff
@@ -386,16 +403,18 @@ set_unique_drug_names <- function(se) {
...
-#' and `Gnumber_3` in parentheses for duplicates.
+#' This function sets the primary drug field(s) in data.frame-like object to be unique 
+#' by appending the secondary drug field(s) in parentheses for duplicates.
+#' By default `DrugName`, `DrugName_2`, and `DrugName_3` are primary drug fields,  
+#' while `Gnumber`, `Gnumber_2`, and `Gnumber_3` are their respective secondary drug fields.
 #'
-#' @param row_data data.table or DFrame with row data
+#' @param dt data.table, data.frame or DFrame with the data
+#' @param primary_names charvec with the names of the primary drug field(s)
+#' @param secondary_name charvec with the name of the secondary drug field(s)
```
[the remark this answers is not in the document]
Comment (gladkia) [reply R/standardize_MAE.R]: Good catch! I've made a fix in 5cd340e79298fcc680fcffb232830d294826134a.

**Pull request merged**